hudi-io/src/test/java/org/apache/hudi/io/hfile/TestHFileReader.java

```diff
@@ -130,14 +139,122 @@ static Stream<Arguments> testArgsReadHFilePointAndPrefixLookup() {
+                new KeyLookUpInfo("hudi-key-000000300a", SEEK_TO_IN_RANGE,
+                    "hudi-key-000000300-abcdefghij", "hudi-value-000000300"),
+                // last key of the block 1
+                new KeyLookUpInfo("hudi-key-000000469-abcdefghij", SEEK_TO_FOUND,
+                    "hudi-key-000000469-abcdefghij", "hudi-value-000000469"),
+                new KeyLookUpInfo("hudi-key-000000469a", SEEK_TO_IN_RANGE,
+                    "hudi-key-000000469-abcdefghij", "hudi-value-000000469"),
+                new KeyLookUpInfo("hudi-key-000000469b", SEEK_TO_IN_RANGE,
+                    "hudi-key-000000469-abcdefghij", "hudi-value-000000469"),
+                // Block 2:
+                // fake first key in block index: hudi-key-00000047
+                // actual first key: hudi-key-000000470-abcdefghij
+                // Lookup key is smaller than actual first key
+                new KeyLookUpInfo("hudi-key-00000047", SEEK_TO_BEFORE_BLOCK_FIRST_KEY,
+                    "hudi-key-000000470-abcdefghij", "hudi-value-000000470"),
+                new KeyLookUpInfo("hudi-key-000000470", SEEK_TO_BEFORE_BLOCK_FIRST_KEY,
+                    "hudi-key-000000470-abcdefghij", "hudi-value-000000470"),
+                // Lookup key is greater than actual first key
+                new KeyLookUpInfo("hudi-key-000000473", SEEK_TO_IN_RANGE,
+                    "hudi-key-000000472-abcdefghij", "hudi-value-000000472"),
+                // Block 29:
+                // fake first key in block index: hudi-key-000006815
+                // actual first key: hudi-key-000006815-abcdefghij
+                new KeyLookUpInfo("hudi-key-000006815-aaa", SEEK_TO_BEFORE_BLOCK_FIRST_KEY,
+                    "hudi-key-000006815-abcdefghij", "hudi-value-000006815"),
+                new KeyLookUpInfo("hudi-key-000006815-aaa", SEEK_TO_BEFORE_BLOCK_FIRST_KEY,
+                    "hudi-key-000006815-abcdefghij", "hudi-value-000006815"),
+                // Backward seekTo within the range [fake first key, actual first key) is OK
+                new KeyLookUpInfo("hudi-key-000006815-aa", SEEK_TO_BEFORE_BLOCK_FIRST_KEY,
+                    "hudi-key-000006815-abcdefghij", "hudi-value-000006815"),
+                new KeyLookUpInfo("hudi-key-000006815", SEEK_TO_BEFORE_BLOCK_FIRST_KEY,
+                    "hudi-key-000006815-abcdefghij", "hudi-value-000006815"),
+                // Backward seekTo before the fake first key is not supported
+                new KeyLookUpInfo("hudi-key-000006814", SEEK_TO_THROW_EXCEPTION, "", ""),
+                new KeyLookUpInfo("hudi-key-000006815-ab", SEEK_TO_BEFORE_BLOCK_FIRST_KEY,
+                    "hudi-key-000006815-abcdefghij", "hudi-value-000006815"),
+                new KeyLookUpInfo("hudi-key-000006815-ac", SEEK_TO_IN_RANGE,
+                    "hudi-key-000006815-abcdefghij", "hudi-value-000006815"),
+                new KeyLookUpInfo("hudi-key-000006816", SEEK_TO_IN_RANGE,
```

Comment:
can we also do one exact match lookup. 
i.e. hudi-key-000006815-abcdefghij

Reply:
Added.